reasoner/resolution/framework/Materialiser.java

```diff
@@ -149,7 +149,7 @@ private Response(Request request, @Nullable Map<Identifier.Variable, Concept> ma
             return sourceRequest().sender();
         }
 
-        public @Nullable Trace trace() {
+        public Trace trace() {
```

Comment:
if these can be null and it isn't too verbose to do this, we should use `Optional` around nullable fields we return

Reply:
applies to other types of messages too

Reply:
Noted, given the big changes coming I'll omit making changes